plugins/rapid7_insightidr/komand_rapid7_insightidr/util/resource_helper.py

```diff
@@ -64,3 +92,42 @@ def resource_request(self, endpoint: str, method: str = "get", params: dict = No
             status_code_message = self._ERRORS.get(response.status_code, self._ERRORS[000])
             self.logger.error(f"{status_code_message} ({response.status_code}): {error}")
             raise PluginException(f"InsightIDR returned a status code of {response.status_code}: {status_code_message}")
+
+    @staticmethod
+    async def _get_log_entries_with_labels(connection: Connection, log_entries: [dict]) -> [dict]:
+        log_ids = set()
+        for log_entry in log_entries:
+            for label in log_entry.get("labels", []):
+                log_ids.add(label.get("id"))
+
+        async with _get_async_session(connection.session.headers) as async_session:
+            tasks: [asyncio.Future] = []
+            for log_id in log_ids:
+                tasks.append(
+                    asyncio.ensure_future(get_label_for_id(log_id=log_id, url=connection.url, session=async_session))
+                )
+
+            labels = await asyncio.gather(*tasks)
+
+        labels_by_id = {}
+        for label in labels:
+            if label:
+                labels_by_id[label.get("id")] = label.get("name")
+
+        new_log_entries: [dict] = []
+        for log_entry in log_entries:
+            new_log_entry = dict(log_entry)
+            new_log_entry["message"] = json.loads(log_entry.get("message", "{}"))
```

Comment:
Please wrap this `json.loads()` call with a try-except for `JSONDecodeError` https://docs.python.org/3/library/json.html?highlight=json%20loads#json.loads